parser.js

```diff
@@ -730,17 +794,77 @@
       }
     }
 
-    // Taken from the Esprima parser.
-    function isIdentifierPart(ch) {
+    function parseIdentifierAtom(check) {
+      var ch = lookahead();
+      var from = pos;
+      if (!check(ch.charCodeAt(0))) return;
+      incr();
+      if (ch === '\\') {
+        var esc = parseRegExpUnicodeEscapeSequence();
+        if (!esc || !check(esc.codePoint)) {
+          bail('Invalid escape sequence', null, from, pos);
+        }
+        return String.fromCodePoint(esc.codePoint);
+      }
+      return ch;
+    }
+
+    function parseIdentifier() {
+      // RegExpIdentifierName ::
+      //      RegExpIdentifierStart
+      //      RegExpIdentifierName RegExpIdentifierContinue
+      //
+      // RegExpIdentifierStart ::
+      //      UnicodeIDStart
+      //      $
+      //      _
+      //      \ RegExpUnicodeEscapeSequence
+      //
+      // RegExpIdentifierContinue ::
+      //      UnicodeIDContinue
+      //      $
+      //      _
+      //      \ RegExpUnicodeEscapeSequence
+      //      <ZWNJ>
+      //      <ZWJ>
```

Comment:
nit: shall we move the grammar for this up at the beginning of the file?

Reply:
Looking over the code once more, other parts of the grammar are also defined locally. Therefore, leaving the grammar definition at the place introduced in this PR.